.github/workflows/pre_release.yaml

```diff
@@ -0,0 +1,40 @@
+name: Canary Release
+
+on:
+  workflow_run:
+    workflows: ["Build and Test"]
+    branches: [main]
+    types: 
+    - completed
+
+jobs:
+  canary-release:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Generate Release Manifests
+      run: make generate-manifests IMAGE=envoyproxy/gateway-dev TAG=latest OUTPUT_DIR=release-artifacts
+
+    - name: Delete canary tag
```

Comment:
is delete needed ?

Reply:
Yes, otherwise the git tag is already existed, we cannot create a new one, then we always have a static tag without newest commits.

Reply:
can we use an action thats actively used by the community ? we need to be careful here

Reply:
Any suggestions?

Reply:
I did not find other better options to do this, deleting tags and release. 😄

Reply:
not atm :),  let me dig around a little, ive seen some repos do this by hand (`git tag -d`)
apart from this, this PR looks good ! this will be super useful for downstream projects and users testing regularly